src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx

```diff
@@ -22,10 +22,11 @@ const propTypes = {
 
 export default function ObligationsByAwardTypeContainer({ fiscalYear, windowWidth }) {
     const [isLoading, setLoading] = React.useState(true);
+    const [isError, setError] = React.useState(false);
     const { toptierCode } = useSelector((state) => state.agencyV2.overview);
-    let isError = false;
 
     if (toptierCode) {
+        let error = false;
```

Comment:
This should set the error. We have access to whether or not there is an error in `isError`. 
`setError(false)`

Reply:
We should also watch the loading here, if the fiscal year is updated we might need to set loading to be true too.

Reply:
I agree, but that causes an infinite rendering loop

Reply:
We can update the component with some additional state, refs, effects and catching api errors to look something like this.

```
export default function ObligationsByAwardTypeContainer({ fiscalYear, windowWidth }) {
    const [categoriesForGraph, setCategoriesForGraph] = useState([]);
    const [detailsForGraph, setDetailsForGraph] = useState([]);
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState(false);
    const obligationsByAwardTypeRequest = useRef(null);
    const { toptierCode } = useSelector((state) => state.agencyV2.overview);

    useEffect(() => () => {
        if (obligationsByAwardTypeRequest.current) {
            obligationsByAwardTypeRequest.current.cancel();
        }
    }, []);

    const getObligationsByAwardType = () => {
        if (obligationsByAwardTypeRequest.current) {
            obligationsByAwardTypeRequest.current.cancel();
        }
        if (error) setError(false);
        if (!loading) setLoading(true);
        obligationsByAwardTypeRequest.current = fetchObligationsByAwardType(toptierCode, fiscalYear);
        obligationsByAwardTypeRequest.current.promise.then((res) => {
            const categories = defaultCategories;
            const details = defaultDetails;
            res.data.results.forEach((d) => {
                switch (d.category) {
                    case 'grants':
                        categories[0].value += d.aggregated_amount;
                        details[0].value = d.aggregated_amount;
                        break;
                    case 'loans':
                        categories[0].value += d.aggregated_amount;
                        details[1].value = d.aggregated_amount;
                        break;
                    case 'direct_payments':
                        categories[0].value += d.aggregated_amount;
                        details[2].value = d.aggregated_amount;
                        break;
                    case 'other':
                        categories[0].value += d.aggregated_amount;
                        details[3].value = d.aggregated_amount;
                        break;
                    case 'idvs':
                        categories[1].value += d.aggregated_amount;
                        details[4].value = d.aggregated_amount;
                        break;
                    case 'contracts':
                        categories[1].value += d.aggregated_amount;
                        details[5].value = d.aggregated_amount;
                        break;
                    default:
                        console.error(`Category name from API not recognized: ${d.category}`);
                        setError(true);
                }
            });
            setCategoriesForGraph(categories);
            setDetailsForGraph(details);
            setLoading(false);
            obligationsByAwardTypeRequest.current = null;
        }).catch((e) => {
            if (!isCancel(e)) {
                setError(true);
                setLoading(false);
                obligationsByAwardTypeRequest.current = null;
                console.error(e);
            }
        });
    };

    useEffect(() => {
        if (toptierCode) {
            getObligationsByAwardType();
        }
    }, [fiscalYear, toptierCode]);


    return (<>
        { loading && <LoadingMessage />}
        { error && <ErrorMessage />}
        { !loading && !error &&
            <ObligationsByAwardType
                outer={categoriesForGraph}
                inner={detailsForGraph}
                windowWidth={windowWidth} />
        }
    </>);
}
```

Updating the categories and details to state should allow react to watch for those updates and rerender the children.

Referencing the request gives us control over it allowing us to cancel any current requests when unmounting.

Moving the request logic to its own function allows us to use an effect to make an api call only when certain data changes.

Catching the response error allows us to ignore and cancel requests we may make and updates the user when there is an api error.

Reply:
Even better ideas; done